Dockerfile

```diff
@@ -10,16 +10,11 @@ COPY requirements.txt .
 COPY requirements_test.txt .
 COPY ocrd_tesserocr ./ocrd_tesserocr
 COPY Makefile .
-RUN apt-get update && \
-    apt-get install -y software-properties-common && \
-    add-apt-repository ppa:alex-p/tesseract-ocr && \
-    apt-get update && \
-    apt-get -y install --no-install-recommends \
-    libtesseract-dev \
-    tesseract-ocr \
+RUN make deps-ubuntu && \
+    apt-get install -y --no-install-recommends \
+    g++ \
```

Comment:
So `g++` is stil needed to build `tesserocr` right?

Reply:
Yes, but I'd have no quarrel adding it to ocrd/core. I don't see a reason why 100MB more or less would matter at this point if it makes implementer's life easier.

Reply:
I was not sure and just tried it. Yes, it is still used, but obviously only for linking:

    x86_64-linux-gnu-g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.7/tesserocr.o -llept -ltesseract -o build/lib.linux-x86_64-3.7/tesserocr.cpython-37m-x86_64-linux-gnu.so

So I am pretty sure this dependency could be dropped with some efforts.

Reply:
> Yes, but I'd have no quarrel adding it to ocrd/core. I don't see a reason why 100MB more or less would matter at this point if it makes implementer's life easier.

I concur. But still I think this dependency should be explicit here – and not just in the Dockerfile, but in the Makefile under `deps-ubuntu`.



> Yes, it is still used, but obviously only for linking:
> 
> ```
> x86_64-linux-gnu-g++ -pthread -shared -Wl,-O1 -Wl,-Bsymbolic-functions -Wl,-z,relro -Wl,-z,relro -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 build/temp.linux-x86_64-3.7/tesserocr.o -llept -ltesseract -o build/lib.linux-x86_64-3.7/tesserocr.cpython-37m-x86_64-linux-gnu.so
> ```
> 
> So I am pretty sure this dependency could be dropped with some efforts.

Yes, but that would require adding assumptions on the way the linker can be called from setuptools (which in turn might be dependent on Python / setuptools versions). Better encapsulate everything and be more robust.

Reply:
So we could remove `g++` here now, but it does not matter really.